File: solutions/java/bob/1/src/main/java/Bob.java
public class Bob {
	public String hey(String msg) {
		msg = msg.trim();

		if (msg.equals(""))
			return "Fine. Be that way!";

		Boolean q = msg.substring(msg.length() - 1).equals("?");
		Boolean s = msg.matches("[^a-z]+");

		if (msg.matches("[^A-z]+")) {
			if (q)
				return "Sure.";
			return "Whatever.";
		}

		if (q && s)
			return "Calm down, I know what I'm doing!";
		if (q)
			return "Sure.";
		if (s)
			return "Whoa, chill out!";

		return "Whatever.";
	}
}

File: solutions/dart/bob/1/lib/bob.dart
class Bob {
  String response(String msg) {
    msg = msg.trim();
    if (msg == "") return "Fine. Be that way!";
    bool s = !RegExp("[a-z]").hasMatch(msg);
    bool q = msg.substring(msg.length - 1) == "?";
    if (!RegExp("[A-z]").hasMatch(msg)) {
      if (q)
        return "Sure.";
      else
        return "Whatever.";
    }

    if (s && q) return "Calm down, I know what I'm doing!";
    if (s) return "Whoa, chill out!";
    if (q) return "Sure.";
    return "Whatever.";
  }
}
